pkg/kubelet/dockertools/exec.go

```diff
@@ -96,10 +94,17 @@ func (*NsenterExecHandler) ExecInContainer(client DockerInterface, container *do
 		if stderr != nil {
 			command.Stderr = stderr
 		}
-
-		err = command.Run()
+		if err := command.Start(); err != nil {
+			return err
+		}
 	}
-
+	if timeout > 0 {
+		t := time.AfterFunc(timeout, func() {
+			command.Process.Kill()
```

Comment:
I think it would be good to log an error ("process timed out..."), and also log any error returned by `Kill`